FWCore/Framework/interface/EDConsumerBase.h

```diff
@@ -267,6 +294,59 @@ namespace edm {
     bool frozen_;
     bool containsCurrentProcessAlias_;
   };
+
+  template <Transition TR>
+  class EDConsumerBaseESAdaptor {
+  public:
+    template <typename TYPE, typename REC>
+    ESGetToken<TYPE, REC> consumes() const {
```

Comment:
Would the function better be non-`const`?
```suggestion
    ESGetToken<TYPE, REC> consumes() {
```
It does call a non-`const` function of the contained pointer. The adaptors for `ConsumesCollector` and `ESConsumesCollector` have non-`const` consumes functions.